pandas/core/frame.py

```diff
@@ -2234,7 +2234,7 @@ def maybe_reorder(
 
         if is_iterator(data):
             if nrows == 0:
-                return cls()
+                return cls(columns=columns)
```

Comment:
Does `index` need to be preserved here as well?

Reply:
Right, to be consistent with empty iterator and `nrows > 0`. I updated the PR.